src/views/user/ResetPassword/ResetPassword.jsx

```diff
@@ -0,0 +1,307 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { PropTypes } from 'prop-types';
+import axiosInstance from '../../../services/requestHandler';
+import resetPasswordActions from '../../../actions/resetPasswordActions';
+import ResetPasswordForm from './ResetPasswordForm';
+import InvalidTokenMessage from './InvalidTokenMessage';
+
+const { setToken, updatePassword } = resetPasswordActions;
+
+/**
+ *
+ * The ResetPassword component renders view to get
+ * user email and hits the route that sends reset
+ * password link to user email
+ * @class ResetPassword
+ * @returns {undefined} this is a class thus does not return anything
+ */
+export class ResetPassword extends React.Component {
+  state = {
+    password: '',
+    confirm: '',
+    disabled: true,
+    isValidToken: false,
+    frontendValidations: {
+      password: [],
+      confirm: [],
+    },
+  };
+
+  validity = {
+    passwordIsValid: false,
+    passwordIsConfirmed: false,
+  };
+
+  frontendValidations = {
+    password: [],
+    confirm: [],
+  };
+
+  /**
+   * componentDidMoutn
+   * @returns {null} null
+   */
+  componentDidMount() {
+    const { location, dispatch, history } = this.props;
+    const resetToken = location.search.substr(7);
+    axiosInstance().get(`/users/reset-password?token=${resetToken}`)
+      .then(() => {
+        dispatch(setToken(resetToken));
+        this.setState({ isValidToken: true });
+        return history.replace(location.pathname);
+      })
+      .catch(() => {
+        this.setState({ isValidToken: false });
+      });
+  }
+
+  /**
+   * Handles validation of the confirm password input
+   * @param {Object} input the target input
+   * @returns {undefined} no value is returned
+   */
+  validateConfirmPassword = (input) => {
+    const { password } = this.state;
+    if (input.name !== 'confirm') return;
+    if (!input.value) {
+      this.validity.passwordIsConfirmed = false;
+      return;
+    }
+    if (input.value !== password) {
+      this.validity.passwordIsConfirmed = false;
+      return;
+    }
+    this.validity.passwordIsConfirmed = true;
+  }
+
+  /**
+   * Handles validation of the password input
+   * @param {Object} input the target input
+   * @returns {undefined} no value
+   */
+  validatePassword = (input) => {
+    if (input.name !== 'password') return;
+    if (!input.value) {
+      this.validity.passwordIsValid = false;
+      return;
+    }
+    const regex = new RegExp(['^(?=.*[\\d])(?=.*[A-Z])(?=.*[a-z])',
```

Comment:
renname regex

Reply:
so I see you having plenty regex everywhere, do you think you can have a utils folder where you have helpers handling all regex cases, instead of doing that in a component

Reply:
I used them only in two instances and both cases handle different patterns. So, they are not reused.